### DISCUSSION TITLE
Server Name or Hostname access

### DISCUSSION THREAD

**Discussion by mustafa519** [uNetworking/uWebSockets.js#995] [General]
Hello,

First, thank you for providing this package to the community, including me. It feels like a game changer for node.js ecosystem. And I'll do my best to bring something cool by using uws.js.

Recently, I started implementing a framework like MVC pattern on top of uws.js. Minimal API was really fun to develop like tiny details let me do anything I want so far.

However, at the current progress, I am kind of stuck. I am trying to proxying the incoming request but, I have only the host header which let me create the full URL of the incoming request. As far I know, host header cannot be trusted in any case.

The question is, how can I have the exact requested full URL which is reliable?

`addServerName` callback is cool and lets me know if the request is not from the host that I haven't set. Still, there is no way to know what is the server name of the incoming request.

Or am I missing something after read ten pages of discussions and all the examples too?

P.S: This is my first experience getting deeper on the HTTP. I appreciate it if you guys would lead me.

Thanks,
Mustafa.

**Comment by uNetworkingAB** [comment]
The server name matches whatever your pattern in the router is. So if you want to catch a specific servername, add a handler for that servername. There is no getter (I think?) in the library yet, but that could also be added like getServerName()

**Reply by mustafa519** [reply]
> The server name matches whatever your pattern in the router is.

Yeah, I catch the server name on `missingServerName` callback however, there is no way to establish context between `missingServerName` and router callback at the runtime. Let me know, if I am wrong.

In most cases, uws is faster than nginx. So, I am trying to get it to work without a wrapper if it's possible until I get stuck.

> but that could also be added like getServerName()

That would be cool!

By the way, where would you get the matched server name to expose the API? I mean, from the host header? Or is there a DNS resolver like architecture in the native uws?

I am only checking the `host` header for now. Not sure if it's the best way. If it doesn't match, I behave like it's localhost and return 400. I hope this is safe when I expose dynamic subdomains.
